### PR TITLE
feat(workspace): add Go workspace (go.work) discovery

### PR DESCRIPTION
**Pull request by Strum355**
## Summary
- Add `GoWorkspace` record for deserializing `go work edit -json` output
- Add `discoverGoWorkspaceModules()` method in `ExhortApi` for discovering `go.mod` manifest paths in Go multi-module workspaces
- Integrate Go workspace detection into `discoverWorkspaceManifests()` (between Gradle and JavaScript in the detection order)

## Test plan
- [x] 4 GoWorkspace deserialization tests (standard output, null Use, empty Use, unknown fields)
- [x] 7 discovery tests (multi-module, nested, single, missing dir, command failure, empty Use, ignore patterns)
- [ ] CI verification (no JDK available locally)

## Dependencies
Depends on #444 (Gradle workspace discovery - TC-4262) for the base branch

## Jira
[TC-4264](https://redhat.atlassian.net/browse/TC-4264)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[TC-4264]: https://redhat.atlassian.net/browse/TC-4264?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by Sourcery

Extend workspace manifest discovery to support Go multi-module workspaces and Gradle multi-project builds, while tightening default ignore patterns and adding comprehensive tests for the new discovery logic.

New Features:
- Detect Go workspaces and discover go.mod manifests using go work edit -json output
- Detect Gradle multi-project builds and collect build.gradle and build.gradle.kts manifests via a custom init script

Enhancements:
- Expand default workspace discovery ignore patterns to include build, target, and .gradle directories

Tests:
- Add Go workspace deserialization tests covering standard, null, empty, and unknown-field cases
- Add Go workspace discovery tests for multi-module, nested, single-module, failure, missing module, and ignore-pattern scenarios
- Add Gradle workspace discovery tests for various project structures, command failures, missing subprojects, and ignore-pattern filtering

### CODE REVIEW

File: src/main/java/io/github/guacsec/trustifyda/impl/ExhortApi.java [hunk 1]
```diff
@@ -930,6 +948,163 @@ private List<Path> discoverCargoManifests(Path workspaceDir, Set<String> ignoreP
...
+    for (String line : raw.split("\n")) {
+      if (!line.startsWith("::DA_PROJECT::")) {
+        continue;
+      }
+      String[] parts = line.split("::");
+      List<String> nonEmpty = new ArrayList<>();
+      for (String part : parts) {
+        if (!part.isEmpty()) {
+          nonEmpty.add(part);
+        }
```
[the remark this answers is not in the document]
Comment (Strum355) [reply src/main/java/io/github/guacsec/trustifyda/impl/ExhortApi.java]: **Classification: Out of scope (belongs to PR #444)**

This review comment targets the Gradle `parseGradleInitScriptOutput` method (lines 1090-1099), which is part of PR #444 (Gradle workspace discovery - TC-4262), not this PR's Go workspace discovery changes. The Gradle code appears in this diff only because this branch is based on #444.

The CRLF concern is valid and should be tracked against PR #444 where the Gradle parsing was introduced.

*This comment was AI-generated by [sdlc-workflow/verify-pr](https://github.com/mrizzi/sdlc-plugins) v0.5.11.*

**Comment by Strum355** [comment]
## Verification Report -- TC-4264: Go workspace (go.work) discovery

### Comment Classification

| Thread | Author | Classification | Action |
|--------|--------|---------------|--------|
| Reviewer's Guide | sourcery-ai (bot) | Informational | No action needed |
| Test Results: 464/464 pass | github-actions | CI feedback | Verified |
| CRLF/whitespace in Gradle parsing | sourcery-ai (review) | Out of scope -- belongs to PR #444 (Gradle discovery) | Replied with classification |

### Scope Check

- **PR title:** `feat(workspace): add Go workspace (go.work) discovery`
- **Jira task:** TC-4264 -- Add Go workspace (go.work) discovery to Java client
- **Branch:** TC-4264
- **Changed files (Go-specific to this PR):**
  - `ExhortApi.java` -- Added `discoverGoWorkspaceModules()` method + Go check in `discoverWorkspaceManifests()`
  - `GoWorkspace.java` -- New JSON model record for `go work edit -json`
  - `WorkspaceUtils.java` -- Fixed glob matching for root-level paths (`**/X/**` pattern)
  - `module-info.java` -- Opens/exports `golang.model` package for Jackson
  - `GoWorkspaceDiscoveryTest.java` -- 11 unit tests (4 deserialization + 7 discovery)
  - 10 test fixture files under `src/test/resources/tst_manifests/workspace/go/`
- **Scope verdict:** All changes are directly related to Go workspace discovery as specified in TC-4264.

### Commits

| Commit | Message | Assessment |
|--------|---------|------------|
| `db1e043` | `feat(workspace): add Go workspace (go.work) discovery` | Core implementation |
| `68de12e` | `fix(workspace): export golang.model to Jackson and fix glob matching` | Fix for module-info and PathMatcher quirk |

Both commits are Co-Authored-By Claude Opus 4.6 as expected.

### CI Status

- **Unit tests:** 464/464 passing
- **Integration tests:** 44/45 passing
  - **1 failure:** `windows-latest / gradle-kotlin` -- `Command 'gradle dependencies' timed out after 120 seconds`. This is a pre-existing Gradle timeout on Windows, unrelated to Go workspace changes.
- **Build:** Passing (Java 21)
- **Sourcery review:** Passing

### Sensitive Patterns

No credentials, secrets, API keys, or tokens found in the diff.

### Acceptance Criteria Verification

| Criteria | Status | Evidence |
|----------|--------|----------|
| `stackAnalysisBatch("/path/to/go-workspace")` discovers all member `go.mod` files | PASS | `discoverWorkspaceManifests_goMultiModule` test verifies 2 modules discovered from go.work |
| `GoWorkspace` record correctly deserializes `go work edit -json` output | PASS | `goWorkspace_deserializesStandardOutput` test + `@JsonIgnoreProperties(ignoreUnknown = true)` |
| Relative paths in `Use` entries correctly resolved | PASS | `discoverWorkspaceManifests_nestedModules` test verifies `./libs/core` and `./libs/util` resolution |
| Members with non-existent directories skipped gracefully | PASS | `discoverWorkspaceManifests_missingModuleDirectory` test: `./existing` found, `./nonexistent` skipped |
| If `go` is not available, returns empty list with logged warning | PASS | `discoverWorkspaceManifests_goCommandFails` test: exit code 1 returns empty list |

### Test Requirements Verification

| Requirement | Status | Test |
|-------------|--------|------|
| Simple workspace: discovers correct number of `go.mod` files | PASS | `discoverWorkspaceManifests_goMultiModule` (2 modules) |
| Missing member directory: skipped | PASS | `discoverWorkspaceManifests_missingModuleDirectory` |
| Empty use list: returns empty | PASS | `discoverWorkspaceManifests_emptyUseList` |
| JSON deserialization test for GoWorkspace record | PASS | 4 tests: standard, null Use, empty Use, unknown fields |

### Additional Tests (beyond requirements)

- `discoverWorkspaceManifests_singleModule` -- single-module workspace
- `discoverWorkspaceManifests_nestedModules` -- deeply nested module paths
- `discoverWorkspaceManifests_ignorePatternFiltering` -- ignore pattern integration

### Code Quality Notes

1. **Pattern follows existing conventions:** `discoverGoWorkspaceModules()` follows the same CLI-invoke-parse-filter pattern as `discoverCargoManifests()`.
2. **GoWorkspace record:** Clean Jackson model with `@JsonIgnoreProperties(ignoreUnknown = true)`, matching the `CargoMetadata` pattern.
3. **WorkspaceUtils fix:** The glob matching fix for root-level `**/X/**` patterns is a legitimate bugfix needed for ignore pattern filtering to work correctly at the workspace root level.
4. **Discovery order:** Go workspace check is correctly placed between Gradle and JavaScript in the detection order, as specified.

### Verdict: PASS

All acceptance criteria and test requirements are met. The single CI failure is an unrelated Gradle timeout on Windows. The implementation is clean, well-tested, and follows existing patterns.

*This comment was AI-generated by [sdlc-workflow/verify-pr](https://github.com/mrizzi/sdlc-plugins) v0.5.11.*

**Review by a-oren — APPROVED** [review]
LGTM
